### ISSUE TITLE
Add rendering for natural=valley labels

### ISSUE THREAD

**Issue by matthijsmelissen** [gravitystorm/openstreetmap-carto#788]
The following issue has been moved over from [trac](https://trac.openstreetmap.org/ticket/2396):

The following are all tags used for common natural features in Iceland (and probably elsewhere) on nodes (along with place=locality to get them to render). They're traditionally rendered as plain text on maps:
- ~~natural=fjord~~
- natural=valley
- ~~natural=peninsula~~
- ~~natural=channel~~
- ~~natural=inlet~~

The list is ordered roughly by how prominent each should be on the map. But of course these things can differ wildly.

**Comment by mkoniecz** [comment]
natural=heath label is now rendered.

**Comment by math1985** [comment]
The tag natural=peninsula has 59 occurrences, natural=channel has 34 occurrences, and natural=inlet has 9 occurrences. That's too little to render them.

**Comment by dieterdreist** [comment]
2015-03-22 1:45 GMT+01:00 math1985 <notifications@github.com>:

> The tag natural=peninsula has 59 occurrences, natural=channel has 34
> occurrences, and natural=inlet has 9 occurrences. That's too little to
> render them.



I don't agree, these are typically very significant features and supporting
them will increase the overall cartographic quality. Also there exist much
fewer of these features in the world than of other features, so maybe those
numbers aren't that bad (refering to channel and peninsula).

**Comment by imagico** [comment]
Note natural=channel, natural=inlet, natural=fjord and natural=sound are all undocumented and it is not clear what exactly distinguishes them from natural=strait and natural=bay.

natural=fjord is the only one with more widespread and consistent use, could be defined as something like 'long and narrow bay formed by glaciers' and would make sense to be rendered together with natural=bay.

**Comment by Tomasz-W** [comment]
Labels to render:
* natural=fjord
* natural=valley
* natural=cape
* natural=ridge

**Comment by hansfn** [comment]
It would help a lot if the initial post was updated with the labels in the comment above and any other labels mentioned in the previous comments. I probably wouldn't have created issue #3148 then :-)

**Comment by meased** [comment]
This may be a question more general to this style than this particular issue, but are there any issues using `text-character-spacing` with all of the languages we render?

Some features (such as valleys, maybe even streams/rivers) would look nice with character separation but I believe some languages rely heavily on ligatures.

Has anyone looked into this before?

**Comment by kocio-pl** [comment]
Ask @sommerluk about such things.

**Comment by sommerluk** [comment]
> Some features (such as valleys, maybe even streams/rivers) would look nice with character separation

Indeed _letter-spacing_ looks nice for features like big valleys or big water bodies. It’s something I would like to have.

>  but I believe some languages rely heavily on ligatures.

The problem are not so much the ligatures. OpenType does a quite good job separating typographic ligatures like “ﬁ” from orthographic ligatures like “æ”: The former should break up when using _letter-spacing_, the latter not.

> are there any issues using `text-character-spacing` with all of the languages we render?

> Has anyone looked into this before?

The problem are cursive scripts. [Quoting myself from another issue comment](https://github.com/gravitystorm/openstreetmap-carto/issues/2692#issuecomment-318041654):

> - Regular/[letter-spacing](https://en.wikipedia.org/wiki/Letter-spacing) works for LGC scripts and probably most other scripts without big problems [except cursive scripts](https://drafts.csswg.org/css-text/#cursive-tracking) like arabic. [Some Mapbox software even disables letter-spacing for Arabic scripts](https://github.com/mapbox/mapbox-gl-js/issues/4208).

Apart from Arabic, there are many Indic scripts that have joined letter forms, [though it seem they use sometimes _letter-spacing_, but the rules are complex and based on syllables, not on letters](https://github.com/harfbuzz/harfbuzz/issues/64#issuecomment-259253470).

The [latest Noto release “phase III”](https://www.google.com/get/noto/) has not only introduced support for many new scripts, but also various font widths that are designed by hand and are a good choice for usage – not only for the Latin script, but also for many other scripts including Arabic and various Indic scripts. Unfortunately, the list of width is

- extracondensed
- condensed
- semicondensed
- normal

There are no “expanded” variants. It’s really a pity, because an “expanded” style would be perfect for our usage. (Anyway, maybe we can make use of one of the condensed styles one day…)

So, in summary I see some risks when using _letter-spacing_, that is difficult to control for complex scripts without breaking rendering – and scripts like Arabic and the Indic scripts together might have 10% or 20% of the population world-wide, which is considerable.

**Comment by meased** [comment]
Thanks for the explanation @sommerluk, very informative.

**Comment by jragusa** [comment]
`natural=fjord` is discouraged on the [wiki page](https://wiki.openstreetmap.org/wiki/Tag:bay%3Dfjord) in favour to `bay=fjord`. Since `natural=bay`is already supported (#3144), we can exclude fjord from the above todo list.

**Comment by jeisenbe** [comment]
That’s right.

We could still add a special rendering for natural=bay with bay=fjord
mapped as a linear way, with the text following the way.

**Comment by Tomasz-W** [comment]
Task list:
- [ ] natural=cape #3148 -> PR https://github.com/gravitystorm/openstreetmap-carto/pull/3452
- [ ] natural=ridge #1148 #2774
- [ ] natural=valley
- [ ] bay=fjord (https://github.com/gravitystorm/openstreetmap-carto/issues/788#issuecomment-431508369)

**Comment by geozeisig** [comment]
Please render the label for **ocean** and **sea** in the same way as bay. For ocean the tag `place=ocean` is used and for a large sea `place=sea` (Do you believe that place is right here?)
For example for the [Mediterranean Sea](https://www.openstreetmap.org/node/305640275) there is a node. I think that's too little. We should use a polygon. But if we use the coastline, even a relation will be overloaded. We should draw a polygon that roughly represents the coastline and whose segments are about 10 km.
But which tag is right here? Or do we use bay or strait?

**Comment by jragusa** [comment]
@geozeisig related to #1982 ?

**Comment by Tomasz-W** [comment]
@geozeisig There is already a ticket for this issue: https://github.com/gravitystorm/openstreetmap-carto/issues/2278

**Comment by Adamant36** [comment]
Anyone have any idea how valleys or ridges could be rendered? I thought there was a PR for ridge rendering that was closed that could be used as an example, but I can't seem to find it. Maybe it was for something else.

**Comment by kocio-pl** [comment]
There was  #1148 and #2774, you can check it.

**Comment by Adamant36** [comment]
Thanks. I looked through #1148 before and didn't see it. Now I notice its issue #2138 though.

**Comment by matkoniecz** [comment]
>  I’ve reverted this change to the wiki.

Thanks, "tagging on ways does not render.” used as a sole reason is a clear case of tagging for render.

**Comment by jragusa** [comment]
For the record: there is a new discussion about peninsula in the [tagging list](https://lists.openstreetmap.org/pipermail/tagging/2018-December/041637.html)

**Comment by eehpcm** [comment]
natural=valley appears to have fallen through the gaps (see the crossed-off items in the first post in this thread and issue #2774).

In my local area there are a lot of named valleys.  I've tagged a few of them with natural=valley but the labels don't render.  In this part of Wales, valley names often serve the same purpose, and are used the same way, as place=neighbourhood in England - they are distinct regions (albeit long and narrow) within a larger named locality and are used by locals to refer to those regions.

I could replace it with place=neighbourhood, but that isn't for use on linear ways and doesn't convey that the place is, in fact, a valley.  I could replace it with place=locality, but over on the tagging mailing list @jeisenbe suggested trying to find more specific tagging to use instead of that.  So here I am, asking if labels can be rendered for place=valley.

**Comment by jeisenbe** [comment]
> A standard text label? There's a tag just for adding a label?

No. I’m sorry that I wasn’t clear.

I meant that the name could be rendered as a horizontal black text label
using the standard font (not bold or italic) centered on the node or the
center of the way. This would similar to the rendering of other landform
features such as natural=cape or place=island.

**Comment by eehpcm** [comment]
@jeisenbe 

>  I meant that the name could be rendered as a horizontal black text label using the standard font (not bold or italic) centered on the node or the center of the way. This would similar to the rendering of other landform features such as natural=cape or place=island.

That would be better than nothing. :)  I don't see a need for rendering the way itself, and it would probably be confusing if it were rendered.  However, I suggest that the label be oriented in the general direction of the way.  See [Cwm Morganau](https://www.openstreetmap.org/edit#map=17/52.05878/-4.60594) in the OS_OpenData_StreetView layer.  There is no need to curve the text along the way (as is done with road and river names), just get the general direction.  OTOH, calculating a "general" direction from a contorted way might be a difficult algorithm to get right, and you already have code for curving labels of roads and rivers.

**Comment by jeisenbe** [comment]
Changed title since natural=valley is the only remaining feature on the list. I believe it would be worth trying a simple label-only rendering for valleys. This would at least prevent tagging them as `place=locality`

**Comment by eehpcm** [comment]
@jeisenbe 

Test cases to see how they look (and how they compare with OS OpenData StreetView):

[Cwm Plysgog](https://www.openstreetmap.org/way/672728569)

[Cwm Rhew-erwyll](https://www.openstreetmap.org/way/617168684)

[Cwm Morgenau](https://www.openstreetmap.org/way/684390284)

The first two should look OK(ish) whether the label is horizontal or follows the valley.  The third is not going to look good (and be misleading) unless the label follows the valley.

**Comment by jeisenbe** [comment]
Thank you for the examples.

> The third is not going to look good (and be misleading) unless the label follows the valley.

Looking at Cwm Morgenau, it won't be much more misleading that the rendering of the linear woodland to the west 

![cwm-morgenau-before-z15](https://user-images.githubusercontent.com/42757252/65370342-9e426c80-dc92-11e9-8822-82eef313a0e8.png)
(the valley is the orange line, the wood has the green text label west of the river)

Note that the `natural=wood` area named Allt Pen-lan (which is to the north and west of the river) is rendered with a label placed near the center, even though the wood is long, narrow and curved. A hand-crafted map would also want to label this with a curve: https://www.openstreetmap.org/way/638346143#map=16/52.0591/-4.6115

While following the OSM way with the label will be better in some cases, a non-oriented label at the center of the valley will work in most cases.

**Comment by eehpcm** [comment]
@jeisenbe 

> Looking at Cwm Morgenau, it won't be much more misleading that the rendering of the linear woodland to the west

Actually, a horizontal label for the valley **would** be more misleading, because woods are rendered objects and valleys are not.  So if you see a patch of green wood with a label, you know the maximum possible extent and shape of the wood.  Maximum possible extent and shape because it may be a named chunk of a larger area of woodland, as [Allt Pen-lan](https://www.openstreetmap.org/way/638346143#map=15/52.0580/-4.6124) is (actually, I've probably made that named chunk extend more to the north than it really does, and if I fixed that the label would render closer to where OS OpenData StreetView puts it).  With a valley, that doesn't render, all you have to go on is the label itself.  The two situations aren't really comparable.

I know you already have code that places labels so they follow ways because you do that with roads and rivers.  I know this means labelling valleys would consume more resources at render time, but at least you don't have to write such code from scratch (I'd hope it's a subroutine/procedure call but copy/paste is still easier than writing from scratch).  Roads and rivers do render, so the label following the way isn't vital, it just makes the map look better.  When labelling an unrendered valley, the label is all you have to go on.

> While following the OSM way with the label will be better in some cases, a non-oriented label at the center of the valley will work in most cases.

Assuming the valley doesn't meander much (which it may), a non-oriented label will work (even for very small values of "work") in only 50% of cases.  It will work for the 50% of cases where the valley deviates no more than 45 degrees from horizontal.  It will fail for the 50% of cases where the valley deviates no more than 45 degrees from vertical.  Even with the most generous definition of "work" it's wrong half the time.

All that said, a non-oriented label is better than no label at all, and I'll settle for it if that's all I can get.  But if you do add the label, people are more likely to use the tag, and I'd expect that eventually you will get some of those people asking for oriented labels.

**Comment by jeisenbe** [comment]
> With a valley, that doesn't render, all you have to go on is the label itself.

That's why it may be a problem to render just the label, oriented along the way. This could encourage mappers to draw "labeling geometries", based on how they want the label to look on openstreetmap.org, rather than based on a verifiable feature on the ground. 

For valleys that contain rivers or streams, usually that would be the lowest spot, so that could be a verifiable place to draw the `natural=valley` way. However, I notice in [your examples](https://github.com/gravitystorm/openstreetmap-carto/issues/788#issuecomment-533548799) that the valley does not exactly follow the waterway. I've seen this commonly in other valleys drawn as ways: the mapper often uses a simplified line, instead of following the thalweg (line of lowest elevation).

Since `natural=valley` ways are not being drawn in a particularly consistent way, we would not want to contribute to this problem and encourage more "mapping for the renderer". But valleys are real, physical geography features, even if it is difficult to map their boundaries or exact course, and by placing a label on  the midpoint of the line, or the node, we can provide a basic idea of the centerpoint of the valley. This is better than mappers using `place=locality` - which can include valleys in it's definition, unfortunately.

If mappers start using `natural=valley` in a more consistent way, for example by attempting to follow the lowest ground in the valley (usually the same as the waterway, if there is one), we could reconsider using those geometries for labeling purposes. 

I'm aware that the rendering will look nicer when using the ways - in many cases, if we just let mappers directly draw the size and location of our labels they would come out better, but we have a responsibility not to push mappers to do the work of renderers for them.

**Comment by eehpcm** [comment]
@jeisenbe 

> That's why it may be a problem to render just the label, oriented along the way. This could encourage mappers to draw "labeling geometries", based on how they want the label to look on openstreetmap.org, rather than based on a verifiable feature on the ground.

You appear to assume that many mappers would wish the label to look like anything other than the course of the valley.  I think that most mappers, like myself, would like the label to follow the valley but accept small imperfections introduced by the rendering process.  I think what is more likely if the label is non-oriented is that they will trim the valley length or course to try to get the label in the centre of the valley (essentially twiddling the line to get the same result as using a node).

> For valleys that contain rivers or streams, usually that would be the lowest spot, so that could be a verifiable place to draw the natural=valley way. However, I notice in your examples that the valley does not exactly follow the waterway. I've seen this commonly in other valleys drawn as ways: the mapper often uses a simplified line, instead of following the thalweg (line of lowest elevation).

I suspect the other mappers did it that way for the same reason I did: that's what the wiki suggests.  The river or stream will almost always follow the thalweg, but the valley is often a good deal wider than the waterway and the waterway meanders within it.  This is why the wiki says "Note that valleys and waterways are not necessarily congruent. Rivers are often meandering. There may be multiple branches of the river, or no river at all. Therefore, you better ignore waterways in U-shaped valleys, and go for the middle line of the valley bottom instead."  That is what I attempted to do, as best I could interpret aerial images and the OpenTopoMap layer.

> Since natural=valley ways are not being drawn in a particularly consistent way, we would not want to contribute to this problem and encourage more "mapping for the renderer".

I am not convinced that they are being drawn inconsistently.  I think it likely that the discrepancy is in your expectation that the valley follow the thalweg and mappers trying to map the centre of the valley as the wiki tells them to.  It's _easier_ to map the thalweg if the waterway is already there, just follow the waterway.  It's more work to try and figure out the centre of the valley.  And, right now, people most definitely aren't tweaking the valley in order to affect label placement because there are no labels.

> If mappers start using natural=valley in a more consistent way, for example by attempting to follow the lowest ground in the valley (usually the same as the waterway, if there is one), we could reconsider using those geometries for labeling purposes.

I believe they already are using them consistently.  I would have taken the easy way out and followed the watercourse had not the wiki persuaded me to put a little more effort in to do the job properly.  But even if I'd decided to avoid the watercourse just because I didn't want to superimpose the valley on it, I'd have placed the valley close by and following a somewhat smoother path.  

Nobody has abused the tag to place a label because there currently are no labels.  Might they do that in the future in order to get around the renderer suppressing a label because of space restrictions?  Maybe, but there are already many other opportunities for them to do things like that if they're so inclined.

**Comment by jeisenbe** [comment]
> It's more work to try and figure out the centre of the valley.

I agree. From my experience trying to map some valleys here in
Indonesia with linear ways, it's quite hard to decide on where the
center of a wide valley should be. Is it the line that is halfway
between the tops of the ridges on each side, or the line at the center
of the flatish part of the valley, or something in between?
Unfortunately this isn't defined.

> Nobody has abused the tag to place a label because there currently are no labels.

The labels are shown in other styles like Opentopomap, which show
contour lines and hillshading. It's more appropriate to show linear
valleys differently in that case, since mappers will see if there is
something clearly wrong, like a line that has been draw without regard
to the actual shape of the valley to just get a certain label
position.

**Comment by eehpcm** [comment]
@jeisenbe 

> Is it the line that is halfway between the tops of the ridges on each side, or the line at the center of the flatish part of the valley, or something in between?

It's an approximation.  The best we can do using aerial imagery and OpenTopoMap.  Complicated by the fact that the valley may contain a wood.  But **everything** we do is an approximation.  Look at the Afon Teifi just north of Cwm Morgenau.  Two years ago it came from NPE and was a very crude approximation to the true course.  Some time in the last 2 years I tweaked it, but I had no idea where the thalweg was so placed it roughly midway between the banks.  Sometime after that I mapped the banks to show the true width.  One day somebody might actually map the thalweg, but until then it's a better approximation than it was with NPE data.  Comparing different aerial imagery shows transitional, rotational and scaling differences, so everything is approximate.

No two mappers would map the same valley exactly the same way.  But most conscientious mappers would come up with approximations that were in broad agreement and that were good enough.

> Unfortunately this isn't defined.

I think that may be a good thing.  It would turn a difficult task (that's a reasonable approximation) to an impossible one (there's no way of knowing the exact centreline to that standard, so I can't map it).

> The labels are shown in other styles like Opentopomap,

I hadn't realized that.  I just checked Cwm Morgenau and can't see why the label is there twice.  I don't think it's something I did wrong, it looks like OpenTopoMap had a brain fart.

> It's more appropriate to show linear valleys differently in that case, since mappers will see if there is something clearly wrong, like a line that has been draw without regard to the actual shape of the valley to just get a certain label position.

I still don't see that as a huge problem.  Sure, I _could_ map a valley, be unhappy with the non-oriented label placement, remap it, check again, remap it,... until I was happy with the label placement.  Or I could just add a place=locality and do a lot less work.  I think the majority of valleys will be mapped as honest attempts to show the true valley, not to place labels.  I also think that if standard carto renders a valley label no differently from that which would be shown by place=locality, more mappers will be tempted to just use place=locality because it's far less work.  In fact, I think if valleys are shown with non-oriented labels, that makes it _more_ likely mappers will tweak the valley (possibly a lot) just to get a better placement when they'd accept an oriented label that looked a little unaesthetic as simply representing reality.

**Comment by mboeringa** [comment]
>  I think the majority of valleys will be mapped as honest attempts to show the true valley, not to place labels.

I have seen few if any problems with mappers mis-using valley tagging on lines and polygons. I agree most are honest attempts to show location of the valley. 

The main problem I see with current tagging sometimes is that people simply replicate the course of a river or main stream running through the the valley, which may not always be appropriate for the valley as a whole.

**Comment by jeisenbe** [comment]
Here are a couple of valleys that I mapped a year or two ago (though I'm not too happy with them now). Attempting to render a text label on the line does not work well:

![iniye-z14](https://user-images.githubusercontent.com/42757252/77251765-a3a95f80-6c93-11ea-86f4-bedac60cb26d.png)
- The valley line should closely follow the river, but this looks bad when rendered on the line.

![iniye-z15](https://user-images.githubusercontent.com/42757252/77251768-a4da8c80-6c93-11ea-83c9-f52e1d5e3369.png)

![lower-baliem-valley-z15](https://user-images.githubusercontent.com/42757252/77251772-ab690400-6c93-11ea-9cdb-9ea31e61e67a.png)

![grand-baliem-valley-z13](https://user-images.githubusercontent.com/42757252/77251775-aefc8b00-6c93-11ea-80a8-2eb3b054fde6.png)

![grand-baliem-valley-z14](https://user-images.githubusercontent.com/42757252/77251779-b02db800-6c93-11ea-984b-3148a06946b4.png)

![grand-baliem-valley-z16](https://user-images.githubusercontent.com/42757252/77251781-b1f77b80-6c93-11ea-9a93-08d266e53ce4.png)
- Here the label might be confused as belonging to the residential area.

Perhaps some combination of simplification of the line, plus stretching out the label to match the length of the way to some extent could work in another style, but probably not in this style which lacks a representation of the terrain.

**Comment by jeisenbe** [comment]
I considered rendering the valley label near the center of the way (or on the point). But this looks odd at high zoom levels, when the valley is large, though it is fine for small valleys.

@imagico, in this case would it be appropriate to stop rendering the labels for very long valleys which extend off the screen, as we would for the text label of a large area?

E.g:
```
  ST_Length(way)/NULLIF(SQRT(!pixel_width!::real*!pixel_height!::real),0) AS way_length,
```
(though it would be better to use scale_denominator)
```
  [feature = 'natural_valley'][zoom >= 12][way_length < 1200] {
```

I know we wouldn't want to show text labels larger or at an earlier zoom level without a representation of the line, but perhaps this is acceptable, since it only is removing the text label when the valley is too long for the usual screen size?

**Comment by eehpcm** [comment]
@jeisenbe 
> Here are a couple of valleys that I mapped a year or two ago (though I'm not too happy with them now). Attempting to render a text label on the line does not work well:

What other maps do (which may be a simple matter of programming here or very, very difficult) is offset the label so it doesn't overlap the watercourse.  It results in a label that isn't positioned exactly, but gives a reasonable indication of the length and geometry of the valley.

**Comment by jeisenbe** [comment]
> offset the label so it doesn't overlap the watercourse.

Generalization is a great idea for map legibility, but another very important goal of this style is mapper feedback, and for that it is important for the precise geometry to be shown. That's why we don't move overlapping roads along rivers. Also the processing would be somewhat computationally expensive, and the more complex code would be harder to edit and maintain

**Comment by imagico** [comment]
Independent of the questions if there is a consistent and verifiable mapping of valleys at the moment i would - like in case of polygons - be very much against labeling linear ways without providing feedback on the geometry and its extent.  That would incentivize drawing of non-verifiable labeling geometries.

**Comment by jeisenbe** [comment]
As mentioned above, my suggest is to render a horizontal text label at the center point of the `natural=valley` way only.

The only possible use of the line length would be to stop rendering valleys which were clearly much larger than the usual screen size. All would start rending at the same zoom level.

If this isn't feasible, we could only render labels on nodes. At least in that case it would incentivize using natural=valley instead of a place=locality node, and would get us a step closer to removing the rendering of place=locality.

**Comment by imagico** [comment]
> As mentioned above, my suggestion is to render a horizontal text label at the center point of the natural=valley way only.

> The only possible use of the line length would be to stop rendering valleys which were clearly much larger than the usual screen size. All would start rending at the same zoom level.

I see - i am doubtful about the usefulness of that regarding the goals of this style.  Even if i imagine that tag was consistently used on linear ways in some form (which is purely wishful thinking) that would only lead to a meaningful labeling at a very small range of zoom levels and what range that is depends a lot on the circumstances.  There are valleys that are in size of the order of a few hundred meters while there are also valleys of a few hundred kilometers.  The range of sensible zoom levels for those two does not even overlap.

**Comment by jeisenbe** [comment]
> there are also valleys of a few hundred kilometers

Fortunately, very few of these are mapped as ways in Openstreetmap. I haven't found any so far in the dozen areas that I have looked at. 

> There are valleys that are in size of the order of a few hundred meters

Most valleys which I have checked are in this range of a few hundred meters to a few kilometers.

Many areas have no `natural=valley` features mapped, e.g. most developing countries which I checked, and flat areas like most of the Netherlands and Delaware. Also most US states and Canadian provinces do not have any valleys mapped yet. 

However, there are some mountainous and hill areas where they are commonly used. 

Of the places I have downloaded, Burgenland in eastern Austria and Hautes Alpes in France have the most valleys, and fortunately they are fairly consistently used there. Some regions of Spain also have a number of valleys, though these are often very coarselly mapped with just 2 to 4 nodes, but I'm still happy to encourage tagging these, rather than using locality for all small unpopulated place names (very common in France and Spain).

**Comment by jeisenbe** [comment]
In these test images I've rendered the valley name label on the center of the line (or on the node) with landform-color (same as peaks/saddles), and rendered a thin landform-color line along the way, if there is one, to show the underlying data. This is not a proposed rendering, but a way to show what the data looks like. Testing at z14 since that would be the earliest to show these features (thoug z15 would be consistent with ridges, aretes and cliffs).

## Hautes Alpes
- 26 ways, all except 1 with a name, no nodes
- Never added to the same feature as waterway=stream, though sometimes it follows a similar geometry

https://www.openstreetmap.org/#map=14/44.4583/6.5547
![valleys-france-14:44 4583:6 5547](https://user-images.githubusercontent.com/42757252/77716821-5bd36100-7022-11ea-91d0-81bf564dc0c7.png)

https://www.openstreetmap.org/#map=15/44.8117/6.7131
![valley2-france-15:44 8117:6 7131](https://user-images.githubusercontent.com/42757252/77716916-8a513c00-7022-11ea-8405-76c35aaf3ff7.png)

https://www.openstreetmap.org/#map=15/44.8117/6.7131
- 5 valleys - one name is blocked by the peak icon (northeast)
![valley2-france-15:44 8117:6 7131](https://user-images.githubusercontent.com/42757252/77717008-bcfb3480-7022-11ea-977c-f4faed68fede.png)

https://www.openstreetmap.org/#map=15/44.6942/5.8519
- Smallest one. This works at z15, but z14 is pushing it
![valley-france-15:44 6942:5 8519](https://user-images.githubusercontent.com/42757252/77717062-ddc38a00-7022-11ea-94b9-258b90a5e8ef.png)

![valley-france-14:44 6942:5 8519](https://user-images.githubusercontent.com/42757252/77717089-e9af4c00-7022-11ea-9020-977bc51c0b02.png)

== Burgenland ==
- 61 ways, no nodes, all but 2 with names. 
- This appears to be a karst region, since many valleys lack a waterway feature. 
- One of the forest polygons is broken in this extract; pretend the background is green.
- Obviously the test rendering doesn't work well with track roads.
https://www.openstreetmap.org/#map=14/47.9775/16.8216
![valleys-2-burgen-14:47 9775:16 8216](https://user-images.githubusercontent.com/42757252/77717279-62160d00-7023-11ea-9606-ea8e7b083978.png)

https://www.openstreetmap.org/#map=14/47.8852/16.5819
![valleys-more-burgen-14:47 8852:16 5819](https://user-images.githubusercontent.com/42757252/77717284-66dac100-7023-11ea-807c-3a10734b02f4.png)

https://www.openstreetmap.org/#map=14/47.8828/16.5008]
![valleys-many-burgen-14:47 8828:16 5008](https://user-images.githubusercontent.com/42757252/77717304-7528dd00-7023-11ea-86f7-d46df15522aa.png)

https://www.openstreetmap.org/#map=14/47.4354/16.2099
![valley-stream-burgen-14:47 4354:16 2099](https://user-images.githubusercontent.com/42757252/77717316-7ce88180-7023-11ea-9e35-a44ff89a3c49.png)

https://www.openstreetmap.org/#map=14/47.3785/16.3751
![valleys-4-burgen-14:47 3785:16 3751](https://user-images.githubusercontent.com/42757252/77717391-ae614d00-7023-11ea-9783-878f2288237e.png)

https://www.openstreetmap.org/#map=15/47.3355/16.3519
![valleys-3-burgenland-15:47 3355:16 3519](https://user-images.githubusercontent.com/42757252/77717405-bd47ff80-7023-11ea-90a8-946b547961e2.png)

== Asturias, Spain ==
- I also checked Extremadura, but there were no valleys there.
- Asturias has 5 ways with natural=valley, all named. 

https://www.openstreetmap.org/#map=14/43.2623/-4.8033
![valley-canal-lilamedu-14:43 2623:-4 8033](https://user-images.githubusercontent.com/42757252/77717648-6262d800-7024-11ea-8949-d0624ef1c622.png)

At z16 lots of place=locality features are visible nearby
![valley-canal-lilamedu-16:43 2623:-4 8033](https://user-images.githubusercontent.com/42757252/77717652-62fb6e80-7024-11ea-9b9a-e657c4fc6fb5.png)

https://www.openstreetmap.org/#map=14/43.1975/-4.7749
- Valle de las moñetas is duplicated as a place=locality node, the intermittent waterway has a different name "riega de las moñetas"
![valley-monetas-14:43 1975:-4 7749](https://user-images.githubusercontent.com/42757252/77717702-79a1c580-7024-11ea-9e5f-6638233ad078.png)

![valley-monetas-16:43 1975:-4 7831](https://user-images.githubusercontent.com/42757252/77717963-05b3ed00-7025-11ea-950f-6f365319260c.png)

https://www.openstreetmap.org/#map=15/43.1951704/-5.2233951
- This valley looks like a pass or saddle too
![foz-valley-15:43 1951704:-5 2233951](https://user-images.githubusercontent.com/42757252/77717987-195f5380-7025-11ea-9299-0594dd59a017.png)

https://www.openstreetmap.org/#map=14/43.3719/-4.8662
- Somewhat roughly mapped
![asturias-valley1-14:43 3719:-4 8662](https://user-images.githubusercontent.com/42757252/77718026-3431c800-7025-11ea-85b1-907772613553.png)

**Comment by jeisenbe** [comment]
There are a few valleys mapped in Tasmania, Wales and Hawaii, though many other English-speaking areas do not have any mapped yet.

== Tasmania ==
- 5 valleys, all mapped as ways but one, all named
- The ways are rather rough

https://www.openstreetmap.org/#map=14/-43.6010/146.8627
- Finally a node
![blowhole-valley-14:-43 6010:146 8627](https://user-images.githubusercontent.com/42757252/77718205-a3a7b780-7025-11ea-8771-d200597ef9ef.png)

https://www.openstreetmap.org/#map=14/-41.6632/145.9525
Marigold valley - rather small, perhaps it is a ravine?
![marigold-valley-14:-41 6632:145 9525](https://user-images.githubusercontent.com/42757252/77718239-bae6a500-7025-11ea-8b3e-ffd5c7d393b2.png)

![marigold-valley-15:-41 6632:145 9525](https://user-images.githubusercontent.com/42757252/77718254-c4700d00-7025-11ea-9236-94aaff2ebfa6.png)

https://www.openstreetmap.org/#map=14/-41.8186/146.3115
- Fun Biblical / Holy Land names in this area
![valley-of-hinom-14:-41 8186:146 3115](https://user-images.githubusercontent.com/42757252/77718301-e10c4500-7025-11ea-91e7-63fee7d3dfae.png)

https://www.openstreetmap.org/#map=14/-41.9200/146.1540
![valley-never-never-14:-41 9200:146 1540](https://user-images.githubusercontent.com/42757252/77718321-eff2f780-7025-11ea-978d-1af867170f78.png)

== Wales ==
- 3 ways, all named

https://www.openstreetmap.org/#map=15/51.9998/-4.6185
![z15-valley-wales-15:51 9998:-4 6185](https://user-images.githubusercontent.com/42757252/77718402-32b4cf80-7026-11ea-89a6-b3b79223de2e.png)

https://www.openstreetmap.org/#map=14/52.0590/-4.6242
- z14 is marginal for these 2 small valleys
![z14-valleys-wales-14:52 0590:-4 6242](https://user-images.githubusercontent.com/42757252/77718425-42341880-7026-11ea-9464-bb820e73b45e.png)

== Hawaii ==
- No nodes, 42 ways, but only 30 named. The 12 unnamed ways are also tagged waterway=*, as are 8 of the named ways - these would not be rendered.
- Lots here are mapped in gullies which might deserve to have a waterway=intermittent mapped as well, based on the aerial imagery, though often the name includes "valley" rather than "gully" or "stream" etc.

https://www.openstreetmap.org/#map=16/23.0581/-161.9168
- These ways are double-tagged as valleys and intermittent streams, though the name includes the word "valley":
![hawaii-nihoa-valleys-streams-16:23 0581:-161 9168](https://user-images.githubusercontent.com/42757252/77718721-eddd6880-7026-11ea-8d15-b1821487d9fa.png)

https://www.openstreetmap.org/#map=15/20.0296/-155.8133
- Small gulches which look like intermittent streams on aerial imagery (there are bridges on the road)
![hawaii-gulches-15:20 0296:-155 8133](https://user-images.githubusercontent.com/42757252/77718748-fe8dde80-7026-11ea-926d-90e01fe3b3ab.png)

https://www.openstreetmap.org/#map=14/21.8934/-160.0998
- All these valleys are fairly nicely mapped. They should also have the intermittent streams mapped, though this has not been done yet. The name appears to refer to the valley not the stream.
![hawaii-niihau-valleys-14:21 8934:-160 0998](https://user-images.githubusercontent.com/42757252/77718791-16656280-7027-11ea-892d-b8f271eb4b93.png)

**Comment by jeisenbe** [comment]
Besides the 2 really big valleys that I mapped here in Papua, the only really large valleys I have found so far are in Norway - partially they appear big due to the high lattitude. Here are there some that really should be shown at z12 or z13.

== Norway, Sogn og Fjordane ==
- One node, 35 ways, all named. One is also a waterway=stream.

https://www.openstreetmap.org/#map=14/61.4989/8.3040
- Mapped as a node, moderate sized per topo map
![norway-valley-node-14:61 4989:8 3040](https://user-images.githubusercontent.com/42757252/77718922-70febe80-7027-11ea-942a-9490e826a0dd.png)

![norway-valley-node-Opentopomap](https://user-images.githubusercontent.com/42757252/77718981-91c71400-7027-11ea-9b2a-87c2a8874701.png)

Most of the ways are 5 to 20 kilometers long (22/35), like these. The very longest is 38 but includes part of a fjord: only 5 are over 20km.

https://www.openstreetmap.org/#map=14/61.4749/8.2782
![norway-valley-way-long-14:61 4749:8 2782](https://user-images.githubusercontent.com/42757252/77719066-b91de100-7027-11ea-81b4-c1c111b43dcc.png)

https://www.openstreetmap.org/#map=14/61.4825/7.9626
![norway-valley-long-14:61 4825:7 9626](https://user-images.githubusercontent.com/42757252/77719083-c044ef00-7027-11ea-97b5-0d1e9da60f1a.png)

https://www.openstreetmap.org/#map=14/61.4640/8.0680
![norway-valley-long-14:61 4640:8 0680](https://user-images.githubusercontent.com/42757252/77719098-c89d2a00-7027-11ea-8d8b-dd4fdc87c7b5.png)

But there are 8 that are less than 5 kilometers long, (about 1/4) eg:
https://www.openstreetmap.org/#map=14/61.9903/6.4627
![norway-short-valley-14:61 9903:6 4627](https://user-images.githubusercontent.com/42757252/77719253-33e6fc00-7028-11ea-8799-156b57db4ffc.png)

**Comment by jeisenbe** [comment]
=== Analysis of all natural=valley ways ===
I downloaded all 11,106 natural=valley ways with name=* tags - ouch, that was 49 megabytes.

2495 have a waterway= tag (mostly waterway=stream), but most of these are waterway=drystream (stream is the next most at 846). Also, 1547 of these are all in one small region of Russia(?) centered on 50.8797426, 40.1522567, most added by User:keder in 2016-17

No other combinations are very significant.

- 95% are shorter than 13 kilometers (only 579 have waylength > 12,000 meters)
- 90% are shorter than 8 kilometers
- 85% are shorter than 5 kilometers

The median `natural=valley` way is 1281 meters long, according to JOSM.

- 95%  are longer than 150 meters
- 90% are longer than 300 meters
- 85% are longer than 400 meters

So a super-majority (70%) are between 0.4 kilometers and 5 kilometers in length, and 80% are 300m to 8km in length

The middle 46% are between 0.5 and 2.0 kilometers: only 2 zoom levels difference.

=== Conclusion ===

I think we can consider rendering these features by focusing on the majority that are a couple hundred meters to a few kilometers in length (and also the small number of nodes should be rendered). 

Unfortunately I do not have any good ideas for a linear representation. Since these are not usually nodes, a point icon (like natural=saddle) would not be appropriate. 

So I am planning to just render the name at the center, though probably in gray rather than landform-color.

**Comment by jeisenbe** [comment]
I believe a gray label is the best option. While I considered something similar to the `landform-color` text labels used for peaks and saddles, that is matching an icon.

As a text label only it might make more sense to match the place=locality, place=island and natural=cape labels, and natural=ridge / natural=arete which are gray.

Options:

*1) Brown labels (based on landform-color)*
![valleys-after-brown-14:44 4596:6 5612](https://user-images.githubusercontent.com/42757252/77728511-fb531c80-703f-11ea-8d54-e3b431489e11.png)

![valley-vs-ridge-france-15:45 0822:6 3171](https://user-images.githubusercontent.com/42757252/77728674-538a1e80-7040-11ea-987e-c8c11bb17b2f.png)


*2) Gray labels, oblique*
![valleys-after-gray-oblique-14:44 4596:6 5612](https://user-images.githubusercontent.com/42757252/77728553-1aea4500-7040-11ea-84c5-43c647e2d349.png)

![martignare-valley-ridge-gray-oblique](https://user-images.githubusercontent.com/42757252/77728707-6a307580-7040-11ea-9511-8afe12377999.png)


*3) Gray labels, standard (book-fonts)*
![valleys-after-gray-book-14:44 4596:6 5612](https://user-images.githubusercontent.com/42757252/77728576-250c4380-7040-11ea-88c2-968fbd1f636a.png)

![martignare-valley-ridge-15-gray-book](https://user-images.githubusercontent.com/42757252/77728727-73214700-7040-11ea-8f89-c215d26de70a.png)

**Comment by jragusa** [comment]
Grey label with standard font would be consistent with ridge names. I also notice that `natural=cape` is labelled in black colour, do we need to have two colours for natural features ?

**Comment by imagico** [comment]
> So a super-majority (70%) are between 0.4 kilometers and 5 kilometers in length, and 80% are 300m to 8km in length

> The middle 46% are between 0.5 and 2.0 kilometers: only 2 zoom levels difference.

It is natural for this kind of feature there are a large number of smaller ones and only very few larger ones.  But that does not mean there are no larger ones.  Like:

https://www.openstreetmap.org/way/374056124
https://www.openstreetmap.org/way/539764751
https://www.openstreetmap.org/way/706934555
https://www.openstreetmap.org/way/539763901

Overall i have strong doubts about the kind of labeling you suggest.  The main problem with the current mapping of natural=valley is that there is no consistency in the geometries.  The suggested labeling would not help changing that because the mapper will get feedback about having done something right (by there appearing a label) but that positive feedback will appear no matter how they draw the line.  At the same time the benefit for the map usefulness seems rather small by placing a point label somewhere along the mapped geometry.

**Comment by jeisenbe** [comment]
So you would only be in favor of labeling these if we also show the line?

I haven't been able to find examples online of linear representations for valleys. Probably in modern maps it is assumed that shading and contour lines are used instead.

Any ideas on how it might be done?

**Comment by imagico** [comment]
> So you would only be in favor of labeling these if we also show the line?

No, even if there was a good way to do that - mapping lacks the consistency for this to make sense.

It seems to me that most of the natural=valley geometries in the database were created to generate a certain result in maps which render labels for those.  This is in particular cases where mappers draw the lines with a deliberate offset to place an offset label.

We could of course do the same and offer mappers the option to directly draw labels in our map.  But i don't think this is the responsible thing to do.

Ultimately it is not within our mandate to tell mappers actively how they are supposed to map valleys.  Therefore i think the only thing we can say at the moment is that the current way these are mapped does not work and we therefore can't support it.

**Comment by eehpcm** [comment]
I just noticed that OpenTopoMap renders valleys.  I'm not saying it's _the_ way to go (especially as I don't know how much computational load their approach requires) but it gives an example of how it could be done.

[z=16](https://opentopomap.org/#map=16/52.05911/-4.60726)
[z=17](https://opentopomap.org/#map=17/52.05910/-4.60729)

I think it looks acceptable.  We might quibble over the text colour for consistency with other features, but apart from that I think it looks OK.

BTW, would a river gorge be considered natural=valley or should it be natural=gorge?  If the latter, would code be added to render natural=gorge if code is added for natural=valley?

**Comment by Ircama** [comment]
I also considered this topic [in the past](https://github.com/gravitystorm/openstreetmap-carto/pull/2138) and as far as I remember the reason why *valley* is not currently rendered in *openstreetmap-carto* is due to the [lack of verifiability]( https://wiki.openstreetmap.org/wiki/Talk:Tag:natural%3Dvalley) in the [valley definition]( https://wiki.openstreetmap.org/wiki/Tag:natural%3Dvalley) (as with other geographical features in OpenStreetMap). For instance, currently its data model does not support intersecting a house with a valley to check whether that house is inside a valley.

Discussions on data models are common to other geographical features in OpenStreetMap, where unfortunately geometry is not yet agreed and related arguments do not provide improvements.

Designing an acceptable geometry for valleys (e.g., whether a valley shall be represented through its [thalweg](https://en.wikipedia.org/wiki/Thalweg) or it is an area) is needed before considering its rendering here. At least this looks the spirit of the maintainers of *openstreetmap-carto*, the official OpenStreetMap rendering, possibly also as an incentive to improve arguable data models of some OpenStreetMap features.

Platforms like [OpenTopoMap](https://opentopomap.org) and [Osmand](https://osmand.net/), which are not official even if remarkable, decided to render valleys and many other geographic features based on current definitions, also in case they are more oriented to directly draw a label with the current renderers than describe the feature.

**Comment by eehpcm** [comment]
> Designing an acceptable geometry for valleys (e.g., whether a valley shall be represented through its thalweg or it is an area) is needed before considering its rendering here. At least this looks the spirit of the maintainers of openstreetmap-carto, the official OpenStreetMap rendering, possibly also as an incentive to improve arguable data models of some OpenStreetMap features.

In this thread there have been many reasons put forward why valleys cannot be rendered, or should not be rendered, or must not be rendered.  One of the arguments was along the lines of "Even if we could rigidly define how to map a valley as a way, there is no good way of rendering it because _reasons."_

I think OpenTopoMap has shown that it is possible to adequately render a valley mapped as a way, even when the label partially obscures the associated river.

Now we can move onto the arguments about accuracy and definitions (and even thalwegs, if you insist).  To which i'll respond that **everything** we map is an approximation and we understand that when better aerial imagery, or better GPS traces or whatever come along then we refine that approximation.  If we insist that we must not map something unless we can map it **perfectly** then we can all give up an go home.

**Comment by Ircama** [comment]
> In this thread there have been many reasons put forward why valleys cannot be rendered, or should not be rendered, or must not be rendered. One of the arguments was along the lines of "Even if we could rigidly define how to map a valley as a way, there is no good way of rendering it because reasons."

I am generally in favor to see names of valleys rendered, together with other mountain features and I have tried to summarize the essence of the issue as I understand it.

**Comment by ppete2** [comment]
I wish to see valleys rendered too. Valleys are an important topological feature, which are in frequent use in talks, newspapers etc.

Btw: On the very top of this issue it's written _"Fixed by #941"_ . Could this be deleted, cause the issue isn't fixed at all.

**Comment by dch0ph** [comment]
I suspect that rendering valley names will look odd in a non-topographic map like Carto.

It can work if you have contours / hill-shading. Here a Carto-derivative where valley names are only rendered at Z11 and Z12:
Z12:
![image](https://github.com/gravitystorm/openstreetmap-carto/assets/20069699/c6f083a3-45fc-4289-82b9-56449f932577)
Z13:
![image](https://github.com/gravitystorm/openstreetmap-carto/assets/20069699/55648103-8cec-4168-895c-bc2b920e0ec1)

The valley name would interfere with the extra detail that appears at Z13.

**Comment by gylfiolafsson** [comment]
I would revive this thread in support of rendering valleys. Here are three reasons:

1. In Iceland, as in many other countries of similar geology, valleys are very highly-ranked as geographical features that are used in everyday speech. Minor issues with rendering should not come in the way of showing these features on maps. 

2. Looking at data on number of tags, we can see that in the 10 years since this thread was started, valleys have grown to be 55.000 worldwide. ![image](https://github.com/user-attachments/assets/39594770-adae-43d0-bb68-f879f6fa4dab) With rendering, this number will increase quickly to cover more valleys. 

3. The tag is approved in the Openstreetmap wiki https://wiki.openstreetmap.org/wiki/Tag:natural%3Dvalley 

Many beautiful renderings are shown here above, indicating that this is perfectly possible in an elegant way.
